dev_docs/key_concepts/api_authorization.mdx

```diff
@@ -0,0 +1,306 @@
+1. **Fine-grained control**: 
+  - Define the exact privileges required to access the route.
+  - Use `requiredPrivileges` to specify privileges with support for complex rules:
+    - **AND rules** using `allRequired`: Requires all specified privileges for access.
+    - **OR rules** using `anyRequired`: Allows access if any one of the specified privileges is met.
+    - **Complex Nested Rules**: Combine both `allRequired` and `anyRequired` for advanced access rules.
+2. **Explicit Opt-out**: Provide a reason for opting out of authorization to maintain transparency.
+3. **Versioned Routes**: Define security configurations for different versions of the same route.
+4. **Improved Documentation with OpenAPI (OAS)**: Automatically generated OAS documentation with the required privileges for each route.
+5. **AuthzResult Object in Route Handlers**: Access the authorization response in route handlers to see which privileges were met.
+
+
+### Configuring Authorization on Routes
+**Before Migration:**
+```ts
+router.get({
+  path: '/api/path',
+  options: {
+    tags: ['access:<privilege_1>', 'access:<privilege_2>'],
+  },
+  ...
+}, handler);
+```
+
+**After Migration:**
+```ts
+router.get({
+  path: '/api/path',
+  security: {
+    authz: {
+      requiredPrivileges: ['<privilege_1>', '<privilege_2>'],
+    },
+  },
+  ...
+}, handler);
+```
+
+### Opting Out of Authorization for Specific Routes
+**Before Migration:**
+```ts
+router.get({
+  path: '/api/path',
+  ...
+}, handler);
+```
+
+**After Migration:**
+```ts
+router.get({
+  path: '/api/path',
+  security: {
+    authz: {
+      enabled: false,
+      reason: 'This route is opted out from authorization because ...',
+    },
+  },
+  ...
+}, handler);
+```
+
+### Classic Router Security Configuration Examples
+
+**Example 1: All Privileges Required**
+Requires `privilege_1` AND both `privilege_2` to access the route.
+```ts
+router.get({
+  path: '/api/path',
+  security: {
+    authz: {
+      requiredPrivileges: ['<privilege_1>', '<privilege_2>'],
+    },
+  },
+  ...
+}, handler);
+```
+
+**Example 2: Any Privileges Required**
+Requires `privilege_1` OR `privilege_2` to access the route.
+```ts
+router.get({
+  path: '/api/path',
+  security: {
+    authz: {
+      requiredPrivileges: [{ anyRequired: ['<privilege_1>', '<privilege_2>'] }],
+    },
+  },
+  ...
+}, handler);
+```
+
+**Example 3: Complex Configuration**
+Requires `privilege_1` AND `privilege_2` AND (`privilege_3` OR `privilege_4`) to access the route.
+```ts
+router.get({
+  path: '/api/path',
+  security: {
+    authz: {
+      requiredPrivileges: [{ allRequired: ['<privilege_1>', '<privilege_2>'], anyRequired: ['<privilege_3>', '<privilege_4>'] }],
+    },
+  },
+  ...
+}, handler);
+```
+
+### Versioned Router Security Configuration Examples
+Versioned router allows different security configurations registered per version, giving flexibility when evolving API security over time.
```

Comment:
```suggestion
Different security configurations can be applied to each version when using the Versioned Router. This allows your authorization needs to evolve in lockstep with your API.
```

@azasypkin I think it'd be nice to include a link here to privilege deprecation documentation (once you have it available). I don't want to give engineers the wrong idea here. I worry this could be easily misunderstood, and lead to insufficient authorization checks as new API versions are introduced.